platform/model/bucket_aggregations/date_histogram.go

```diff
@@ -222,7 +222,21 @@ func (query *DateHistogram) generateSQLForCalendarInterval() model.Expr {
 }
 
 func (query *DateHistogram) getKey(row model.QueryResultRow) int64 {
-	return row.Cols[len(row.Cols)-2].Value.(int64)
+	value := row.Cols[len(row.Cols)-2].Value
```

Comment:
Seems that these 2 functions are the same. Please extract one helper function and use it in both places